File: src/test/java/com/weeth/domain/user/test/fixture/UserTestFixture.java
package com.weeth.domain.user.test.fixture;

import com.weeth.domain.user.domain.entity.User;
import com.weeth.domain.user.domain.entity.enums.Role;
import com.weeth.domain.user.domain.entity.enums.Status;

public class UserTestFixture {

	public static User createActiveUser1() {
		return User.builder()
			.name("적순")
			.email("test1@test.com")
			.status(Status.ACTIVE)
			.build();
	}

	public static User createActiveUser1(Long id) {
		return User.builder()
			.id(id)
			.name("적순")
			.email("test1@test.com")
			.status(Status.ACTIVE)
			.build();
	}

	public static User createActiveUser2() {
		return User.builder()
			.name("적순2")
			.email("test2@test.com")
			.status(Status.ACTIVE)
			.build();
	}

	public static User createActiveUser2(Long id) {
		return User.builder()
			.id(id)
			.name("적순2")
			.email("test2@test.com")
			.status(Status.ACTIVE)
			.build();
	}

	public static User createWaitingUser1() {
		return User.builder()
			.name("순적")
			.email("test2@test.com")
			.status(Status.WAITING)
			.build();
	}

	public static User createWaitingUser1(Long id) {
		return User.builder()
			.id(id)
			.name("순적")
			.email("test2@test.com")
			.status(Status.WAITING)
			.build();
	}

	public static User createWaitingUser2() {
		return User.builder()
			.name("순적2")
			.email("test3@test.com")
			.status(Status.WAITING)
			.build();
	}

	public static User createWaitingUser2(Long id) {
		return User.builder()
			.id(id)
			.name("순적2")
			.email("test3@test.com")
			.status(Status.WAITING)
			.build();
	}

    public static User createAdmin(Long id) {
        return User.builder()
                .id(id)
                .name("적순")
                .email("admin@test.com")
                .status(Status.ACTIVE)
                .role(Role.ADMIN)
                .build();
    }

}

File: src/test/kotlin/com/weeth/domain/user/fixture/UserTestFixture.kt
package com.weeth.domain.user.fixture

import com.weeth.domain.user.domain.entity.User
import com.weeth.domain.user.domain.enums.Status
import com.weeth.domain.user.domain.vo.Email
import org.springframework.test.util.ReflectionTestUtils

object UserTestFixture {
    fun createActiveUser1(id: Long = 0L): User =
        User(
            name = "적순",
            email = Email.from("test1@test.com"),
            status = Status.ACTIVE,
        ).applyId(id)

    fun createActiveUser2(id: Long = 0L): User =
        User(
            name = "적순2",
            email = Email.from("test2@test.com"),
            status = Status.ACTIVE,
        ).applyId(id)

    fun createWaitingUser1(id: Long = 0L): User =
        User(
            name = "순적",
            email = Email.from("test2@test.com"),
            status = Status.WAITING,
        ).applyId(id)

    fun createWaitingUser2(id: Long = 0L): User =
        User(
            name = "순적2",
            email = Email.from("test3@test.com"),
            status = Status.WAITING,
        ).applyId(id)

    fun createRegisteredUser(id: Long = 0L): User =
        User(
            name = "등록완료",
            email = Email.from("registered@test.com"),
            status = Status.ACTIVE,
        ).apply {
            agreeTerms(termsAgreed = true, privacyAgreed = true)
        }.applyId(id)

    fun createAdmin(id: Long = 0L): User =
        User(
            name = "적순",
            email = Email.from("admin@test.com"),
            status = Status.ACTIVE,
        ).applyId(id)

    private fun User.applyId(id: Long): User =
        apply {
            if (id != 0L) ReflectionTestUtils.setField(this, "id", id)
        }
}
